parsl/app/futures.py

```diff
@@ -76,6 +102,30 @@ def filename(self):
         """Filepath of the File object this datafuture represents."""
         return self.filepath
 
+    @property
+    def uuid(self):
+        """UUID of the File object this datafuture represents."""
+        return self.file_obj.uuid
+
+    @property
+    def timestamp(self):
+        """Timestamp when the future was marked done."""
```

Comment:
this is coming from the files st_ctime, I think? which is the first "last modification time" that was observed by Parsl for this file? not coming from future completion with the python process?

Reply:
Yes, the timestamp is meant to be related to the file, not the future. With the change in code from above it is the only "last modification time" Parsl observes.